### PR TITLE
Update password route

### PR DESCRIPTION
**Pull request by Chareeef**
- Implement and test 'PUT /me/update_password'
- Distinguish likes (list of usernames or user_id) from number_of_likes (integer)

### CODE REVIEW

File: db/db_manager.py [hunk 1]
```diff
@@ -137,28 +137,28 @@ def update_user_info(
...
-    ) -> Optional[ObjectId]:
+            old_password: str,
+            new_password: str
+    ) -> Tuple[Dict[str, str], int]:
         """ Update the user's password as hash """
         users = self._db['users']
         user = users.find_one({'_id': ObjectId(user_id)})
 
         if not user:
-            return None
```
Comment (tommy457) [inline db/db_manager.py:148]: Leave it as None the dict and status code should be handled in the routes

File: db/db_manager.py [hunk 2]
```diff
@@ -137,28 +137,28 @@ def update_user_info(
...
+            return {'error': 'user not found'}, 400
 
         if not check_hash_password(user['password'], old_password):
-            return None
```
Comment (tommy457) [inline db/db_manager.py:151]: Leave it as None the dict and status code should be handled in the routes

File: db/db_manager.py [hunk 3]
```diff
@@ -137,28 +137,28 @@ def update_user_info(
...
+            return {'error': 'wrong old password'}, 400
 
         new_hashed_password = hash_pass(new_password)
         try:
             users.find_one_and_update(
                 {'_id': ObjectId(user_id)},
                 {'$set': {'password': new_hashed_password}}
             )
-            return user_id
```
Comment (tommy457) [inline db/db_manager.py:159]: Leave it as user_id the dict and status code should be handled in the routes

File: db/db_manager.py [hunk 4]
```diff
@@ -137,28 +137,28 @@ def update_user_info(
...
+            return {'success': 'password updated'}, 201
         except Exception as e:
-            return None
```
Comment (tommy457) [inline db/db_manager.py:161]: Leave it as None the dict and status code should be handled in the routes

**Review by tommy457 — CHANGES_REQUESTED** [review]

File: db/db_manager.py (hunk 1 above)
Comment (Chareeef) [reply db/db_manager.py:148]: Fixed

**Review by Chareeef — COMMENTED** [review]
I returned descriptive integers from db to know the appropriate response to send from the route

**Pull request merged**